scheduler.go

```diff
@@ -106,6 +106,11 @@ func (s *Scheduler) scheduleNextRun(job *Job) {
 
 	durationToNextRun := s.durationToNextRun(job)
 	job.nextRun = job.lastRun.Add(durationToNextRun)
+
+	// option remove the job's in the scheduler after its last execution
+	if job.runConfig.toRemove && (job.runConfig.maxRuns - job.runCount) == 1 {
+			s.RemoveByReference(job)
```

Comment:
the indentation here is too much 
```suggestion
		s.RemoveByReference(job)
```

Reply:
Also, we are already doing this check [here](https://github.com/go-co-op/gocron/blob/5ed8230dfb9bdf0beb0649cf27c69479dcdc9873/scheduler.go#L459). Would it make more sense to do the cleanup there?
@JohnRoesler @Streppel

Reply:
Yes i think that makes more sense. Move this logic inside `shouldRun` - so when it's checked if a job should run, it also just removes it if it should be removed based on the `toRemove` parameter`.